docs/adr/adr_003_bundling_collection_data.md

```diff
@@ -0,0 +1,209 @@
+
+Requirements
+- Total amount of time to download, extract, and process, must not take longer than fetching attachments individually
+- Handling the bundle in the client should not bloat the main thread or impact user experience in general
+- Feature must not get in the way of legacy clients
+- Implementation should be resilient to race conditions on attachment downloads. Unpacking the bundle can overwrite attachments that had been previously downloaded individually, and individual downloads can overwrite unbundled ones
+- Bundle format should be easily readable by client (third-party libraries are hard to ship in Gecko)
+
+References
+- [The intermediates download code](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/security/manager/ssl/RemoteSecuritySettings.sys.mjs#335-365) that fetches attachments in bulk, and stores state in records
+- [Download of crlite files](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/security/manager/ssl/RemoteSecuritySettings.sys.mjs#620-637)
+- [Download of all regions files](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/security/manager/ssl/RemoteSecuritySettings.sys.mjs#335-365)
+- [Download of all addons suggestions files](https://searchfox.org/mozilla-central/rev/8ec3cc0472ad4f51b254728d024b696eaba82ba0/browser/components/urlbar/private/AddonSuggestions.sys.mjs#87-101)
+
+### Metrics To Measure Before/After
+We want to measure CPU usage, total bandwidth used, duration of syncronization on fresh profiles, and battery usage for full synchronization. We should be able to gather most of this by running power and performance profiles on different hardware.
+
+Hardware to include:
+- Mac Desktop/Laptop
+- Linux Desktop/Laptop
+- Windows Desktop/Laptop
+- Android Mobile
+- iOS Mobile (does it use remote-settings?)
+
+
+## Non Goals
+- Support diffs or delta bundles for clients that already have local data
+- Reduce client performance (ex: we shouldn't increase power usage for clients)
+
+## Assumptions
+- We will find a way for Gecko clients to extract at least one archive format (very likely using third-party library)
+
+## Decision Drivers
+- Keep costs low
+    - With millions of clients, the cost of shipping data is always a concern
+- Server complexity & performance
+    -Monitoring and operating should remain straightforward (ex: limit tech and deployment complexity if possible)
+    -Performance of the service should not be affected
+    - The code in place to support this feature should not involve complex components, libraries, or architecture (eg. to support resumable downloads and range requests)
+- Long-term Maintenance
+- Clients complexity
+    - How complicated it is to know which bundle URL to pull
+
+## Considered Options
+1. Build bundles on attachment upload/removal
+2. Build bundles on Cloud storage event
```

Comment:
If I understand correctly, both options 1 and 2 (re)build bundles on attachment upload/removal. Option 1 handles this in the request / response cycle, and option 2 offloads the work to a cloud function. 

If that's the case, maybe we can include `Build bundles on attachment upload/removal` in both option names. For example:

```suggestion
1. Synchronously build bundles on attachment upload/removal (in request / response cycle)
2. Aynchronously build bundles on attachment upload/removal (via Cloud Storage event)
```